### ISSUE TITLE
shop backend

### ISSUE THREAD

**Issue by jhjacobs81** [linuxkettle/AppShop#4]
Preferably the backend runs on Debian 10. 

- Would we need an elastic search database, or will something simple like postgress do the job? :) 
- What language will the backend be written in?
- How do we make it scalable?

**Comment by kettle-7** [comment]
> Preferably the backend runs on Debian 10.
>- Would we need an elastic search database, or will something simple like postgress do the job? :)
>- What language will the backend be written in?
>- How do we make it scalable?

Wouldn't a simple node server do?

**Comment by jhjacobs81** [comment]
for the first 25 apps, i guess so.. but what about afterwards? i'm in for the long term ;-) how scalable is node?

i have issues with the word "simple" ;-) things like these are never simple.. if you go for simple now, it'll bite your ass later when there's more apps and heavier load?

**Comment by kettle-7** [comment]
I don't know about node scalability. But *for now*, and for the next while when we are still waiting for people to come onboard, we just need something that works, most of the requests will be us testers doing the testing thing.

**Comment by jhjacobs81** [comment]
true.. but at the same time i'm worying that going with node now will eventually prevent us from moving to "the professional thing".

Do you program in any language?

**Comment by kettle-7** [comment]
> Do you program in any language?

I program in a few, and have not many personal preferences. I like Python, but its webserver capabilities pail in comparison to JavaScript, which I also like, but if there's a GUI I like QML, but in QML if you want to extend it with new types you need C++, which is an absolute mess, worse than C, which is still bad, so I like Vala, but that's too close to the GNOME stack and pulls in a lot of weight through GLib and GTK. I can do Java, but don't like it much. I did C# for a while but it's not great on Linux and lacks a cross-platform UI toolkit for frontends.

I don't know!

**Comment by jhjacobs81** [comment]
so.. maybe do a mix? serverbackend in python, serverfrontend in javascript? 

i do some Ruby(on Rails) and ive recently taken up learning Go, i think the latter would be ideal for a backend but i dont know enough of it to have something working within the next few weeks :)

**Comment by kettle-7** [comment]
> ive recently taken up learning Go, i think the latter would be ideal for a backend but i dont know enough of it to have something working within the next few weeks :)

Same here. Go is interesting, but hard to learn.

> so.. maybe do a mix? serverbackend in python, serverfrontend in javascript?

Sounds good. I'm not sure which is faster (we want speed, right?), but they're both interpreted.

**Comment by jhjacobs81** [comment]
We want speed, stability and durability :)

We should pick something that could last for a looong time :) Maybe Go IS the way to Go (pun intended)? it has webserver capability's build in, it compiles into a single binary so updating multiple servers wouldn't be much of a problem (could be automated easily), it has speed..

**Comment by kettle-7** [comment]
Possibly. Just don't drag me into C or C++ land, the land of cavemen

**Comment by jhjacobs81** [comment]
Well no.. idealy we come up with a solution that works for the both of us, so i can help you :) and since Go is the only common denominator, i thought this would be "easy" for both of us :) 

AND.. in the long run i want to write a webdav/settingssync server in Go.. help my users get away from Apple but without the worries of maintaining a linux install..

**Comment by kettle-7** [comment]
This reminds me of the 'thin client' system, but different. If there were a cloud storage service to sync /home, that would be very useful to a very large group of people. I used a private Github repository for this but it got too big.

**Comment by kettle-7** [comment]
> it has webserver capability's build in

That will provide a huge advantage. Definitely the way to Go.

**Comment by jhjacobs81** [comment]
I thought so ;-)
